File: VeroApp/Models/Consultant.swift
import Foundation

struct Consultant: Identifiable, Codable {
    let id: String
    var name: String
    var email: String
    var cpf: String
    var phone: String
    var status: ConsultantStatus
    var registrationDate: Date
    var sponsorCode: String?
    var address: Address
    var consultantCode: String

    enum ConsultantStatus: String, Codable {
        case pending  = "PENDENTE"
        case active   = "ATIVO"
        case inactive = "INATIVO"
        case suspended = "SUSPENSO"

        var label: String { rawValue.capitalized }

        var color: String {
            switch self {
            case .active:    return "success"
            case .pending:   return "secondary"
            case .inactive:  return "subtext"
            case .suspended: return "error"
            }
        }
    }

    var firstName: String { name.components(separatedBy: " ").first ?? name }
}

File: VeroFlutter/lib/models/consultant.dart
import 'address.dart';

enum ConsultantStatus {
  pending('PENDENTE'),
  active('ATIVO'),
  inactive('INATIVO'),
  suspended('SUSPENSO');

  final String label;
  const ConsultantStatus(this.label);
}

class Consultant {
  final String id;
  final String name;
  final String email;
  final String cpf;
  final String phone;
  final ConsultantStatus status;
  final DateTime registrationDate;
  final String? sponsorCode;
  final Address address;
  final String consultantCode;

  const Consultant({
    required this.id,
    required this.name,
    required this.email,
    required this.cpf,
    required this.phone,
    required this.status,
    required this.registrationDate,
    this.sponsorCode,
    required this.address,
    required this.consultantCode,
  });

  String get firstName => name.split(' ').first;
}
